### PR TITLE
HBASE-23092 Make the RM tooling in dev-tools/create-release generic

### PR DESCRIPTION
**Pull request by saintstack**
Make the scripts generic. Set PROJECT environment variable
and away you go; e.g. set it to hbase-operator-tools or to
hbase-thirdparty.

This commit includes a bunch of bugfixes and miscellaneous
that came of trying to use the scripts making RCs.

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| :blue_heart: |  reexec  |   0m 39s |  Docker mode activated.  |
||| _ Prechecks _ |
| :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| :blue_heart: |  shelldocs  |   0m  0s |  Shelldocs was not available.  |
| :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 37s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 11s |  Maven dependency ordering for patch  |
| :broken_heart: |  shellcheck  |   0m  4s |  The patch generated 101 new + 120 unchanged - 50 fixed = 221 total (was 170)  |
| :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
||| _ Other Tests _ |
| :blue_heart: |  asflicense  |   0m  0s |  ASF License check generated no output?  |
|  |   |   2m 31s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-671/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/671 |
| Optional Tests | dupname asflicense shellcheck shelldocs |
| uname | Linux f1590910ce7b 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-671/out/precommit/personality/provided.sh |
| git revision | master / c1a476c492 |
| shellcheck | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-671/1/artifact/out/diff-patch-shellcheck.txt |
| Max. process+thread count | 52 (vs. ulimit of 10000) |
| modules | C:  U:  |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-671/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) shellcheck=0.7.0 |
| Powered by | Apache Yetus 0.11.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by busbey** [comment]
Can we make the project a parameter instead of an env variable?

**Comment by saintstack** [comment]
Yes. Good idea.

**Comment by saintstack** [comment]
@busbey  Addressed your comments sir.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| :blue_heart: |  reexec  |   0m 35s |  Docker mode activated.  |
||| _ Prechecks _ |
| :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| :blue_heart: |  shelldocs  |   0m  1s |  Shelldocs was not available.  |
| :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 38s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 10s |  Maven dependency ordering for patch  |
| :broken_heart: |  shellcheck  |   0m  5s |  The patch generated 67 new + 121 unchanged - 49 fixed = 188 total (was 170)  |
| :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
||| _ Other Tests _ |
| :blue_heart: |  asflicense  |   0m  0s |  ASF License check generated no output?  |
|  |   |   2m 28s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-671/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/671 |
| Optional Tests | dupname asflicense shellcheck shelldocs |
| uname | Linux 2f06839866dc 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-671/out/precommit/personality/provided.sh |
| git revision | master / 52176184df |
| shellcheck | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-671/2/artifact/out/diff-patch-shellcheck.txt |
| Max. process+thread count | 52 (vs. ulimit of 10000) |
| modules | C:  U:  |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-671/2/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) shellcheck=0.7.0 |
| Powered by | Apache Yetus 0.11.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| :blue_heart: |  reexec  |   0m 35s |  Docker mode activated.  |
||| _ Prechecks _ |
| :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| :blue_heart: |  shelldocs  |   0m  0s |  Shelldocs was not available.  |
| :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 36s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 10s |  Maven dependency ordering for patch  |
| :broken_heart: |  shellcheck  |   0m  4s |  The patch generated 67 new + 121 unchanged - 49 fixed = 188 total (was 170)  |
| :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
||| _ Other Tests _ |
| :blue_heart: |  asflicense  |   0m  0s |  ASF License check generated no output?  |
|  |   |   2m 25s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-671/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/671 |
| Optional Tests | dupname asflicense shellcheck shelldocs |
| uname | Linux 3dbe5530e772 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-671/out/precommit/personality/provided.sh |
| git revision | master / 52176184df |
| shellcheck | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-671/3/artifact/out/diff-patch-shellcheck.txt |
| Max. process+thread count | 52 (vs. ulimit of 10000) |
| modules | C:  U:  |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-671/3/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) shellcheck=0.7.0 |
| Powered by | Apache Yetus 0.11.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| :blue_heart: |  reexec  |   0m 36s |  Docker mode activated.  |
||| _ Prechecks _ |
| :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| :blue_heart: |  shelldocs  |   0m  0s |  Shelldocs was not available.  |
| :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 35s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 10s |  Maven dependency ordering for patch  |
| :broken_heart: |  shellcheck  |   0m  4s |  The patch generated 52 new + 114 unchanged - 56 fixed = 166 total (was 170)  |
| :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
||| _ Other Tests _ |
| :blue_heart: |  asflicense  |   0m  0s |  ASF License check generated no output?  |
|  |   |   2m 25s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-671/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/671 |
| Optional Tests | dupname asflicense shellcheck shelldocs |
| uname | Linux 38d86584fc9c 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-671/out/precommit/personality/provided.sh |
| git revision | master / 52176184df |
| shellcheck | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-671/4/artifact/out/diff-patch-shellcheck.txt |
| Max. process+thread count | 52 (vs. ulimit of 10000) |
| modules | C:  U:  |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-671/4/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) shellcheck=0.7.0 |
| Powered by | Apache Yetus 0.11.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| :blue_heart: |  reexec  |   1m 53s |  Docker mode activated.  |
||| _ Prechecks _ |
| :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| :blue_heart: |  shelldocs  |   0m  0s |  Shelldocs was not available.  |
| :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 36s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 11s |  Maven dependency ordering for patch  |
| :broken_heart: |  shellcheck  |   0m  4s |  The patch generated 34 new + 110 unchanged - 60 fixed = 144 total (was 170)  |
| :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
||| _ Other Tests _ |
| :blue_heart: |  asflicense  |   0m  0s |  ASF License check generated no output?  |
|  |   |   3m 44s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-671/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/671 |
| Optional Tests | dupname asflicense shellcheck shelldocs |
| uname | Linux 2fda6043bcc1 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-671/out/precommit/personality/provided.sh |
| git revision | master / 52176184df |
| shellcheck | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-671/5/artifact/out/diff-patch-shellcheck.txt |
| Max. process+thread count | 50 (vs. ulimit of 10000) |
| modules | C:  U:  |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-671/5/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) shellcheck=0.7.0 |
| Powered by | Apache Yetus 0.11.0 https://yetus.apache.org |


This message was automatically generated.

File: dev-support/create-release/do-release-docker.sh [hunk 1]
```diff
@@ -112,7 +118,7 @@ done
 
 GPG_KEY_FILE="$WORKDIR/gpg.key"
 fcreate_secure "$GPG_KEY_FILE"
-$GPG --passphrase $GPG_PASSPHRASE --export-secret-key --armor "$GPG_KEY" > "$GPG_KEY_FILE"
+"$GPG" --passphrase "$GPG_PASSPHRASE" --export-secret-key --armor "$GPG_KEY" > "$GPG_KEY_FILE"
```
Comment (petersomogyi) [inline dev-support/create-release/do-release-docker.sh]: Quoting gpg caused problem previusly. See [#33](https://github.com/apache/hbase-operator-tools/pull/33).

File: dev-support/create-release/do-release-docker.sh [hunk 2]
```diff
@@ -152,13 +158,13 @@ EOF
...
+  echo "JAVA_HOME=/opt/hbase-java" >> "$ENVFILE"
   JAVA_VOL="--volume $JAVA:/opt/hbase-java"
 fi
 
 echo "Building $RELEASE_TAG; output will be at $WORKDIR/output"
 docker run -ti \
   --env-file "$ENVFILE" \
   --volume "$WORKDIR:/opt/hbase-rm" \
-  $JAVA_VOL \
+  "$JAVA_VOL" \
```
Comment (petersomogyi) [inline dev-support/create-release/do-release-docker.sh]: Same problem here. See [#34](https://github.com/apache/hbase-operator-tools/pull/34)

File: dev-support/create-release/release-build.sh [hunk 3]
```diff
@@ -295,53 +255,64 @@ if [[ "$1" == "publish-release" ]]; then
...
   # this must have .asc, and .sha1 - it really doesn't like anything else there
-  for file in $(find hbase* -type f)
+  for file in $(find $modules -type f)
   do
     if [[ "$file" == *.asc ]]; then
       continue
     fi
     if [ ! -f $file.asc ]; then
-      echo $GPG_PASSPHRASE | $GPG --passphrase-fd 0 --output $file.asc \
+      echo "$GPG_PASSPHRASE" | "$GPG" --passphrase-fd 0 --output "$file.asc" \
```
Comment (petersomogyi) [inline dev-support/create-release/release-build.sh]: Probably the same problem here.

File: dev-support/create-release/vote.tmpl [hunk 4]
```diff
@@ -22,7 +23,8 @@ Artifacts were signed with the ${GPG_KEY} key which can be found in:
 
  https://dist.apache.org/repos/dist/release/hbase/KEYS
 
-To learn more about Apache HBase, please see http://hbase.apache.org/
+ To learn more about apache ${PROJECT_TEXT}, please see
```
Comment (petersomogyi) [inline dev-support/create-release/vote.tmpl]: s/apache/Apache

File: dev-support/create-release/vote.tmpl [hunk 5]
```diff
@@ -1,13 +1,14 @@
-Please vote on this Apache HBase release candidate (RC), ${RELEASE_VERSION}.
+Please vote on this apache ${PROJECT_TEXT} release candidate,
```
Comment (petersomogyi) [inline dev-support/create-release/vote.tmpl]: s/apache/Apache

File: dev-support/create-release/vote.tmpl [hunk 6]
```diff
@@ -1,13 +1,14 @@
...
+${PROJECT}-${RELEASE_TAG}
 
 The VOTE will remain open for at least 72 hours.
 
-[ ] +1 Release this package as Apache HBase ${RELEASE_VERSION}
+[ ] +1 Release this package as apache ${PROJECT_TEXT} ${RELEASE_VERSION}
```
Comment (petersomogyi) [inline dev-support/create-release/vote.tmpl]: s/apache/Apache

File: dev-support/create-release/release-util.sh [hunk 7]
```diff
@@ -269,43 +275,115 @@ function init_mvn {
...
+  MAVEN_OPTS="${MAVEN_OPTS}" ${MVN} --settings $tmp_settings install assembly:single -DskipTests \
+    -Dcheckstyle.skip=true "${PUBLISH_PROFILES}" -Dmaven.repo.local="${tmp_repo}"
+
+  # Check there is a bin gz output. The build may not produce one: e.g. hbase-thirdparty.
+  f_bin_tgz="./${PROJECT}-assembly/target/${basename}*-bin.tar.gz"
+  if test -f "${f_bin_tgz}"; then
+    cp "${f_bin_tgz}" ..
+    cd .. || exit
+    for i in "${basename}"*-bin.tar.gz; do
+      echo "$GPG_PASSPHRASE" | "$GPG" --passphrase-fd 0 --armour --output "$i.asc" --detach-sig "$i"
```
Comment (petersomogyi) [inline dev-support/create-release/release-util.sh]: Do not quote GPG

File: dev-support/create-release/release-util.sh [hunk 8]
```diff
@@ -269,43 +275,115 @@ function init_mvn {
...
-  cd ./${yetus}
+  cd ./${yetus} || exit
   ./bin/releasedocmaker -p HBASE --fileversions -v ${release_version} -l --sortorder=newer --skip-credits
   # First clear out the changes written by previous RCs.
   sed -i -e "/^## Release ${release_version}/,/^## Release/ {//!d; /^## Release ${release_version}/d;}" \
-    ${hbase}/CHANGES.md
-  sed -i -e "/^# HBASE  ${release_version} Release Notes/,/^# HBASE/{//!d; /^# HBASE  ${release_version} Release Notes/d;}" \
-    ${hbase}/RELEASENOTES.md
+    ${project}/CHANGES.md || true
+  sed -i -e "/^# ${project} ${release_version} Release Notes/,/^# ${project}/{//!d; /^# ${project} ${release_version} Release Notes/d;}" \
```
Comment (petersomogyi) [inline dev-support/create-release/release-util.sh]: This does not work like this. Check [#35](https://github.com/apache/hbase-operator-tools/pull/35).
Rerunning an RC will duplicate the release notes. This happened with the latest hbase-thirdparty 3.1.0RC1.

File: dev-support/create-release/do-release-docker.sh (hunk 1 above)
Comment (saintstack) [reply dev-support/create-release/do-release-docker.sh]: Thanks @petersomogyi

**Comment by saintstack** [comment]
Thank you @petersomogyi  for review. I was thinking it risky addressing shellcheck complaint w/o doing rull re-run. Thanks for interceding. I copied from hbase-operator-tools but should have done and update first.

**Comment by saintstack** [comment]
Last push addresses @petersomogyi  feedback.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| :blue_heart: |  reexec  |   0m 35s |  Docker mode activated.  |
||| _ Prechecks _ |
| :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| :blue_heart: |  shelldocs  |   0m  0s |  Shelldocs was not available.  |
| :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 37s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 10s |  Maven dependency ordering for patch  |
| :broken_heart: |  shellcheck  |   0m  6s |  The patch generated 34 new + 111 unchanged - 59 fixed = 145 total (was 170)  |
| :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
||| _ Other Tests _ |
| :blue_heart: |  asflicense  |   0m  0s |  ASF License check generated no output?  |
|  |   |   2m 25s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-671/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/671 |
| Optional Tests | dupname asflicense shellcheck shelldocs |
| uname | Linux 98f17fe08720 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-671/out/precommit/personality/provided.sh |
| git revision | master / 2ebdcbc3b0 |
| shellcheck | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-671/6/artifact/out/diff-patch-shellcheck.txt |
| Max. process+thread count | 50 (vs. ulimit of 10000) |
| modules | C:  U:  |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-671/6/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) shellcheck=0.7.0 |
| Powered by | Apache Yetus 0.11.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| :blue_heart: |  reexec  |   0m 32s |  Docker mode activated.  |
||| _ Prechecks _ |
| :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| :blue_heart: |  shelldocs  |   0m  1s |  Shelldocs was not available.  |
| :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 36s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 11s |  Maven dependency ordering for patch  |
| :broken_heart: |  shellcheck  |   0m  5s |  The patch generated 38 new + 107 unchanged - 63 fixed = 145 total (was 170)  |
| :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
||| _ Other Tests _ |
| :blue_heart: |  asflicense  |   0m  1s |  ASF License check generated no output?  |
|  |   |   2m 22s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-671/7/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/671 |
| Optional Tests | dupname asflicense shellcheck shelldocs |
| uname | Linux 5611bb4e8aba 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-671/out/precommit/personality/provided.sh |
| git revision | master / 2ebdcbc3b0 |
| shellcheck | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-671/7/artifact/out/diff-patch-shellcheck.txt |
| Max. process+thread count | 52 (vs. ulimit of 10000) |
| modules | C:  U:  |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-671/7/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) shellcheck=0.7.0 |
| Powered by | Apache Yetus 0.11.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| :blue_heart: |  reexec  |   0m 33s |  Docker mode activated.  |
||| _ Prechecks _ |
| :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| :blue_heart: |  shelldocs  |   0m  0s |  Shelldocs was not available.  |
| :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 36s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 11s |  Maven dependency ordering for patch  |
| :broken_heart: |  shellcheck  |   0m  5s |  The patch generated 34 new + 111 unchanged - 59 fixed = 145 total (was 170)  |
| :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
||| _ Other Tests _ |
| :blue_heart: |  asflicense  |   0m  0s |  ASF License check generated no output?  |
|  |   |   2m 23s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-671/8/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/671 |
| Optional Tests | dupname asflicense shellcheck shelldocs |
| uname | Linux 22999a00f992 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-671/out/precommit/personality/provided.sh |
| git revision | master / 2ebdcbc3b0 |
| shellcheck | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-671/8/artifact/out/diff-patch-shellcheck.txt |
| Max. process+thread count | 52 (vs. ulimit of 10000) |
| modules | C:  U:  |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-671/8/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) shellcheck=0.7.0 |
| Powered by | Apache Yetus 0.11.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**